diff -up dolphin-emu-3.5/CMakeTests/FindSFML.cmake~ dolphin-emu-3.5/CMakeTests/FindSFML.cmake
--- dolphin-emu-3.5/CMakeTests/FindSFML.cmake~	2013-02-20 00:38:34.000000000 +0100
+++ dolphin-emu-3.5/CMakeTests/FindSFML.cmake	2013-11-18 11:57:56.150949832 +0100
@@ -24,7 +24,7 @@ endif()
 
 # find the SFML include directory
 find_path(SFML_INCLUDE_DIR SFML/Config.hpp
-          PATH_SUFFIXES include
+          PATH_SUFFIXES include/sfml1
           PATHS
           ~/Library/Frameworks
           /Library/Frameworks
@@ -87,11 +87,11 @@ set(FIND_SFML_LIB_PATHS ~/Library/Framew
 foreach(FIND_SFML_COMPONENT ${SFML_FIND_COMPONENTS})
     string(TOLOWER ${FIND_SFML_COMPONENT} FIND_SFML_COMPONENT_LOWER)
     string(TOUPPER ${FIND_SFML_COMPONENT} FIND_SFML_COMPONENT_UPPER)
-    set(FIND_SFML_COMPONENT_NAME sfml-${FIND_SFML_COMPONENT_LOWER}${FIND_SFML_LIB_SUFFIX})
+    set(FIND_SFML_COMPONENT_NAME sfml-${FIND_SFML_COMPONENT_LOWER}${FIND_SFML_LIB_SUFFIX}-1.6)
 
     # no suffix for sfml-main, it is always a static library
     if(FIND_SFML_COMPONENT_LOWER STREQUAL "main")
-        set(FIND_SFML_COMPONENT_NAME sfml-${FIND_SFML_COMPONENT_LOWER})
+        set(FIND_SFML_COMPONENT_NAME sfml-${FIND_SFML_COMPONENT_LOWER}-1.6)
     endif()
 
     # debug library
diff -up dolphin-emu-3.5/CMakeLists.txt~ dolphin-emu-3.5/CMakeLists.txt
--- dolphin-emu-3.5/CMakeLists.txt~	2013-11-18 11:58:43.000000000 +0100
+++ dolphin-emu-3.5/CMakeLists.txt	2013-11-18 12:00:41.120408310 +0100
@@ -417,6 +417,7 @@ if(NOT ${CMAKE_SYSTEM_NAME} MATCHES "Dar
 endif()
 if(SFML_FOUND AND NOT SFML_VERSION_MAJOR) # SFML 1.x doesn't define SFML_VERSION_MAJOR
        message("Using shared SFML")
+       include_directories( /usr/include/sfml1 )
 else()
        message("Using static SFML ${SFML_FIND_VERSION_MAJOR}.${SFML_FIND_VERSION_MINOR} from Externals")
        add_subdirectory(Externals/SFML)
diff -up dolphin-emu-3.5/Source/Core/Core/CMakeLists.txt~ dolphin-emu-3.5/Source/Core/Core/CMakeLists.txt
--- dolphin-emu-3.5/Source/Core/Core/CMakeLists.txt~	2013-02-20 00:38:35.000000000 +0100
+++ dolphin-emu-3.5/Source/Core/Core/CMakeLists.txt	2013-11-18 12:14:15.844970552 +0100
@@ -191,7 +191,7 @@ set(SRCS	Src/ActionReplay.cpp
 			Src/PowerPC/JitCommon/JitCache.cpp
 			Src/PowerPC/JitCommon/Jit_Util.cpp)
 
-set(LIBS bdisasm inputcommon videoogl videosoftware sfml-network)
+set(LIBS bdisasm inputcommon videoogl videosoftware sfml-network-1.6)
 
 if(WIN32)
 	set(SRCS ${SRCS} Src/HW/BBA-TAP/TAP_Win32.cpp Src/stdafx.cpp
diff -up dolphin-emu-3.5/Source/Core/DolphinWX/CMakeLists.txt~ dolphin-emu-3.5/Source/Core/DolphinWX/CMakeLists.txt
--- dolphin-emu-3.5/Source/Core/DolphinWX/CMakeLists.txt~	2013-02-20 00:38:35.000000000 +0100
+++ dolphin-emu-3.5/Source/Core/DolphinWX/CMakeLists.txt	2013-11-18 12:14:11.445091807 +0100
@@ -6,7 +6,7 @@ set(LIBS	core
 			common
 			audiocommon
 			z
-			sfml-network
+			sfml-network-1.6
 			${GTK2_LIBRARIES}
 			${OPENGL_LIBRARIES}
 			${XRANDR_LIBRARIES}
