diff -Nru dolphin-emu-3.0.orig/Source/Core/Common/Src/CommonFuncs.h dolphin-emu-3.0/Source/Core/Common/Src/CommonFuncs.h
--- dolphin-emu-3.0.orig/Source/Core/Common/Src/CommonFuncs.h	2012-06-02 20:09:12.652636536 -0400
+++ dolphin-emu-3.0/Source/Core/Common/Src/CommonFuncs.h	2012-06-02 20:09:43.822657738 -0400
@@ -21,6 +21,7 @@
 #ifdef _WIN32
 #define SLEEP(x) Sleep(x)
 #else
+#include <unistd.h>
 #define SLEEP(x) usleep(x*1000)
 #endif
 
