diff -Naur dolphin-emu-3.0.orig/CMakeLists.txt dolphin-emu-3.0/CMakeLists.txt
--- dolphin-emu-3.0.orig/CMakeLists.txt	2012-04-05 11:47:21.171186877 -0500
+++ dolphin-emu-3.0/CMakeLists.txt	2012-04-05 12:49:37.599949558 -0500
@@ -446,8 +446,13 @@
 endif()
 
 if(NOT ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-	include_directories(Externals/CLRun/include)
-	add_subdirectory(Externals/CLRun)
+	if(USE_EXTERNAL_CLRUN)
+		find_library(CL clrun)
+		include_directories("${CLRUN_INCLUDE_PATH}")
+	else(USE_EXTERNAL_CLRUN)
+		include_directories(Externals/CLRun/include)
+		add_subdirectory(Externals/CLRun)
+	endif(USE_EXTERNAL_CLRUN)
 endif()
 
 option(DISABLE_WX "Disable wxWidgets (use CLI interface)" OFF)
