diff -rupN dolphin-emu-3.5.orig/CMakeLists.txt dolphin-emu-3.5/CMakeLists.txt
--- dolphin-emu-3.5.orig/CMakeLists.txt	2013-02-19 18:38:35.000000000 -0500
+++ dolphin-emu-3.5/CMakeLists.txt	2013-02-19 18:58:51.099823833 -0500
@@ -500,7 +500,7 @@ if(NOT DISABLE_WX)
 			)
 		message("Found wxWidgets version ${wxWidgets_VERSION}")
 		if(UNIX AND NOT APPLE)
-			set(wxMIN_VERSION "2.9.3")
+			set(wxMIN_VERSION "2.8.12")
 		else()
 			set(wxMIN_VERSION "2.9.4")
 		endif()
diff -rupN dolphin-emu-3.5.orig/Source/Core/DolphinWX/Src/Frame.cpp dolphin-emu-3.5/Source/Core/DolphinWX/Src/Frame.cpp
--- dolphin-emu-3.5.orig/Source/Core/DolphinWX/Src/Frame.cpp	2013-02-19 18:38:35.000000000 -0500
+++ dolphin-emu-3.5/Source/Core/DolphinWX/Src/Frame.cpp	2013-02-19 19:17:59.323425042 -0500
@@ -753,7 +753,7 @@ void CFrame::OnGameListCtrl_ItemActivate
 
 bool IsHotkey(wxKeyEvent &event, int Id)
 {
-	return (event.GetKeyCode() != WXK_NONE &&
+	return (event.GetKeyCode() &&
 	        event.GetKeyCode() == SConfig::GetInstance().m_LocalCoreStartupParameter.iHotkey[Id] &&
 	        event.GetModifiers() == SConfig::GetInstance().m_LocalCoreStartupParameter.iHotkeyModifier[Id]);
 }
diff -rupN dolphin-emu-3.5.orig/Source/Core/DolphinWX/Src/GameListCtrl.cpp dolphin-emu-3.5/Source/Core/DolphinWX/Src/GameListCtrl.cpp
--- dolphin-emu-3.5.orig/Source/Core/DolphinWX/Src/GameListCtrl.cpp	2013-02-19 18:38:35.000000000 -0500
+++ dolphin-emu-3.5/Source/Core/DolphinWX/Src/GameListCtrl.cpp	2013-02-19 19:23:14.454606065 -0500
@@ -593,9 +593,9 @@ void CGameListCtrl::ScanForISOs()
 			SplitPath(rFilenames[i], NULL, &FileName, NULL);
 
 			// Update with the progress (i) and the message
-			dialog.Update(i, wxString::Format(_("Scanning %s"),
-				wxString(FileName.c_str(), *wxConvCurrent).c_str()));
-			if (dialog.WasCancelled())
+			bool Cont = dialog.Update(i,
+				wxString::Format(_("Scanning %s"), wxString(FileName.c_str(), *wxConvCurrent).c_str()));
+			if (!Cont)
 				break;
 
 			std::auto_ptr<GameListItem> iso_file(new GameListItem(rFilenames[i]));
diff -rupN dolphin-emu-3.5.orig/Source/Core/DolphinWX/Src/ISOProperties.cpp dolphin-emu-3.5/Source/Core/DolphinWX/Src/ISOProperties.cpp
--- dolphin-emu-3.5.orig/Source/Core/DolphinWX/Src/ISOProperties.cpp	2013-02-19 18:38:35.000000000 -0500
+++ dolphin-emu-3.5/Source/Core/DolphinWX/Src/ISOProperties.cpp	2013-02-19 19:27:50.417130313 -0500
@@ -740,11 +740,7 @@ void CISOProperties::ExportDir(const cha
 	{
 		dialog.SetTitle(wxString::Format(wxT("%s : %d%%"), dialogTitle.c_str(),
 			(u32)(((float)(i - index[0]) / (float)(index[1] - index[0])) * 100)));
-		
-		dialog.Update(i, wxString::Format(_("Extracting %s"),
-			wxString(fst[i]->m_FullPath, *wxConvCurrent).c_str()));
-
-		if (dialog.WasCancelled())
+		if (!dialog.Update(i, wxString::Format(_("Extracting %s"), wxString(fst[i]->m_FullPath, *wxConvCurrent).c_str())))
 			break;
 
 		if (fst[i]->IsDirectory())
diff -rupN dolphin-emu-3.5.orig/Source/Core/DolphinWX/Src/Main.cpp dolphin-emu-3.5/Source/Core/DolphinWX/Src/Main.cpp
--- dolphin-emu-3.5.orig/Source/Core/DolphinWX/Src/Main.cpp	2013-02-19 18:38:35.000000000 -0500
+++ dolphin-emu-3.5/Source/Core/DolphinWX/Src/Main.cpp	2013-02-19 19:40:17.136618692 -0500
@@ -114,38 +114,38 @@ bool DolphinApp::OnInit()
 	wxCmdLineEntryDesc cmdLineDesc[] =
 	{
 		{
-			wxCMD_LINE_SWITCH, "h", "help",
-			"Show this help message",
+			wxCMD_LINE_SWITCH, wxS("h"), wxS("help"),
+			wxS("Show this help message"),
 			wxCMD_LINE_VAL_NONE, wxCMD_LINE_OPTION_HELP
 		},
 		{
-			wxCMD_LINE_SWITCH, "d", "debugger",
-			"Opens the debugger",
+			wxCMD_LINE_SWITCH, wxS("d"), wxS("debugger"),
+			wxS("Opens the debugger"),
 			wxCMD_LINE_VAL_NONE, wxCMD_LINE_PARAM_OPTIONAL
 		},
 		{
-			wxCMD_LINE_SWITCH, "l", "logger",
-			"Opens the logger",
+			wxCMD_LINE_SWITCH, wxS("l"), wxS("logger"),
+			wxS("Opens the logger"),
 			wxCMD_LINE_VAL_NONE, wxCMD_LINE_PARAM_OPTIONAL
 		},
 		{
-			wxCMD_LINE_OPTION, "e", "exec",
-			"Loads the specified file (DOL,ELF,GCM,ISO,WAD)",
+			wxCMD_LINE_OPTION, wxS("e"), wxS("exec"),
+			wxS("Loads the specified file (DOL,ELF,GCM,ISO,WAD)"),
 			wxCMD_LINE_VAL_STRING, wxCMD_LINE_PARAM_OPTIONAL
 		},
 		{
-			wxCMD_LINE_SWITCH, "b", "batch",
-			"Exit Dolphin with emulator",
+			wxCMD_LINE_SWITCH, wxS("b"), wxS("batch"),
+			wxS("Exit Dolphin with emulator"),
 			wxCMD_LINE_VAL_NONE, wxCMD_LINE_PARAM_OPTIONAL
 		},
 		{
-			wxCMD_LINE_OPTION, "V", "video_backend",
-			"Specify a video backend",
+			wxCMD_LINE_OPTION, wxS("V"), wxS("video_backend"),
+			wxS("Specify a video backend"),
 			wxCMD_LINE_VAL_STRING, wxCMD_LINE_PARAM_OPTIONAL
 		},
 		{
-			wxCMD_LINE_OPTION, "A", "audio_emulation",
-			"Low level (LLE) or high level (HLE) audio",
+			wxCMD_LINE_OPTION, wxS("A"), wxS("audio_emulation"),
+			wxS("Low level (LLE) or high level (HLE) audio"),
 			wxCMD_LINE_VAL_STRING, wxCMD_LINE_PARAM_OPTIONAL
 		},
 		{
@@ -237,9 +237,9 @@ bool DolphinApp::OnInit()
 
 	if (selectAudioEmulation)
 	{
-		if (audioEmulationName == "HLE")
+		if (audioEmulationName == wxS("HLE"))
 			SConfig::GetInstance().m_LocalCoreStartupParameter.bDSPHLE = true;
-		else if (audioEmulationName == "LLE")
+		else if (audioEmulationName == wxS("LLE"))
 			SConfig::GetInstance().m_LocalCoreStartupParameter.bDSPHLE = false;
 	}
 
diff -rupN dolphin-emu-3.5.orig/Source/Core/DolphinWX/Src/NetWindow.cpp dolphin-emu-3.5/Source/Core/DolphinWX/Src/NetWindow.cpp
--- dolphin-emu-3.5.orig/Source/Core/DolphinWX/Src/NetWindow.cpp	2013-02-19 18:38:35.000000000 -0500
+++ dolphin-emu-3.5/Source/Core/DolphinWX/Src/NetWindow.cpp	2013-02-19 19:43:23.706012655 -0500
@@ -29,6 +29,7 @@
 
 #define NETPLAY_TITLEBAR	"Dolphin NetPlay"
 
+DEFINE_EVENT_TYPE(wxEVT_THREAD)
 BEGIN_EVENT_TABLE(NetPlayDiag, wxFrame)
 	EVT_COMMAND(wxID_ANY, wxEVT_THREAD, NetPlayDiag::OnThread)
 END_EVENT_TABLE()
diff -rupN dolphin-emu-3.5.orig/Source/Core/DolphinWX/Src/WXInputBase.cpp dolphin-emu-3.5/Source/Core/DolphinWX/Src/WXInputBase.cpp
--- dolphin-emu-3.5.orig/Source/Core/DolphinWX/Src/WXInputBase.cpp	2013-02-19 18:38:35.000000000 -0500
+++ dolphin-emu-3.5/Source/Core/DolphinWX/Src/WXInputBase.cpp	2013-02-19 19:46:24.058458372 -0500
@@ -43,7 +43,7 @@ const wxString WXKeyToString(int keycode
 		case WXK_CLEAR:				return _("Clear");
 		case WXK_SHIFT:				return wxT("Shift");
 		case WXK_ALT:				return wxT("Alt");
-		case WXK_RAW_CONTROL:		return _("Control");
+		case WXK_CONTROL:			return _("Control");
 #ifdef __WXOSX__
 		case WXK_COMMAND:			return _("Command");
 #endif
